cli/cli-client/src/main/java/org/infinispan/cli/commands/CommandInputLine.java

```diff
@@ -24,35 +24,76 @@ public CommandInputLine arg(String name, String value) {
       return this;
    }
 
-   public CommandInputLine option(String name, boolean value) {
-      this.options.put(name, Boolean.toString(value));
+   public CommandInputLine optionalArg(String name, Object value) {
+      if (value != null) {
+         this.arguments.put(name, value);
+      }
       return this;
    }
 
-   public CommandInputLine optionalArg(String name, String value) {
+   public CommandInputLine option(String name, String value) {
       if (value != null) {
-         this.arguments.put(name, value);
+         this.options.put(name, value);
+      }
+      return this;
+   }
+
+   public CommandInputLine option(String name, Integer value) {
+      if (value != null) {
+         this.options.put(name, Integer.valueOf(value));
```

Comment:
Guessing you wanted the argument to be `int`? If not then the `valueOf` is unneeded.

Reply:
Right